src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.inline.hpp

```diff
@@ -157,13 +157,6 @@ inline void ShenandoahHeapRegion::increase_live_data_gc_words(size_t s) {
 
 inline void ShenandoahHeapRegion::internal_increase_live_data(size_t s) {
   size_t new_live_data = Atomic::add(&_live_data, s, memory_order_relaxed);
-#ifdef ASSERT
```

Comment:
We have not observed violation of this assert during testing.  However, it appears unreliable in that _live_data increases monotonically under Atomic volatile math, whereas used() increases monotonically under heap lock, which we do not hold at the point of this assertion.

It is possible that another allocating thread increases both used() and _live_data.  We will see the increase in _live_data because _live_data changes are volatile Atomic.  However, we may not see the increase in used() because we did not acquire the heap lock and the value of _top() that contributes to calculation of used() is not volatile.  This situation can lead to an assertion failure.